sdk/lib/rtl/heap.c

```diff
@@ -1289,6 +1293,13 @@ RtlCreateHeap(ULONG Flags,
         Flags &= HEAP_CREATE_VALID_MASK;
     }
 
+#ifdef ENABLE_SYSTEMWIDE_DPH
+    if (!Addr)
+    {
+        Flags |= HEAP_FLAG_PAGE_ALLOCS;
+    }
+#endif
+
```

Comment:
This should not be needed. Look at RtlpPageHeapCreate implementation.

Reply:
Just setting `FLG_HEAP_PAGE_ALLOCS` on `NtGlobalFlag` should be enough, no need for all these hacks everywhere?

Reply:
NtGlobalFlag is per application, so it would be tedious to enable it globally.

Reply:
There is also a global `NtGlobalFlag`, that all applications inherit

Reply:
Right, it's in Session Manager\GlobalFlag Somehow I missed it.

@SergeGautherie maybe setting it in regtest.cmd would do what you want to achieve.

Reply:
> Just setting `FLG_HEAP_PAGE_ALLOCS` on `NtGlobalFlag` should be enough, no need for all these hacks everywhere?

Thanks for the hint.
I tried to do it on NTOS:EX and NTDLL:LDR, but that did not succeed.
Could you tell me where/how to do that exactly?

--

> regtest.cmd

That might be a good idea for running tests, but it is not the system-wide case (from ISO start).